drake/systems/plants/collision/FCLModel.cpp

```
@@ -0,0 +1,421 @@
+            elements[id]->getGeometry());
+        bvh_shape = newFCLSphereShape(sphere, true);
+      } break;
+      case DrakeShapes::CYLINDER: {
+        const auto cylinder = static_cast<const DrakeShapes::Cylinder&>(
+            elements[id]->getGeometry());
+        bvh_shape = newFCLCylinderShape(cylinder, true);
+      } break;
+      case DrakeShapes::MESH: {
+        const auto mesh =
+            static_cast<const DrakeShapes::Mesh&>(elements[id]->getGeometry());
+        // I only see points.  I cannot find triangles.
+        // bt_shape = newBulletMeshShape(mesh, true);
+        // bt_shape_no_margin = newBulletMeshShape(mesh, false);
+        cerr << "Warning: FCLMesh not implemented." << endl;
+      } break;
+      case DrakeShapes::MESH_POINTS: {
+        const auto mesh = static_cast<const DrakeShapes::MeshPoints&>(
+            elements[id]->getGeometry());
+        // bt_shape = newBulletMeshPointsShape(mesh, true);
+        // bt_shape_no_margin = newBulletMeshPointsShape(mesh, false);
+        cerr << "Warning: FCLMeshPoints not implemented." << endl;
+      } break;
+      case DrakeShapes::CAPSULE: {
+        const auto capsule = static_cast<const DrakeShapes::Capsule&>(
+            elements[id]->getGeometry());
+        bvh_shape = newFCLCapsuleShape(capsule, true);
+        cerr << "Warning: FCLCapsule not implemented." << endl;
+      } break;
+      default:
+        cerr << "Warning: Collision elements[id] has an unknown type "
+             << elements[id]->getShape() << endl;
+        throw unknownShapeException(elements[id]->getShape());
+        break;
+    }
+    if (bvh_shape) {
+      this->fclElements.insert(
+          make_pair(id, unique_ptr<BVHModel<OBBRSS>>(bvh_shape)));
+    }
+  }
+  return id;
+}
+
+vector<PointPair> FCLModel::potentialCollisionPoints(bool use_margins) {
+  ResultCollector c;
+  // TODO: Not Implemented
+  cerr << "Warning: FCLModel::potentialCollisionPoints not implemented."
```

in general I would throw an exception if not implemented yet